extensions/machinepools/machinepools.go

```diff
@@ -290,3 +298,27 @@ func MatchRoleToPool(poolRole string, allRoles []Roles) int {
 	logrus.Warn("unable to match pool to role, likely missing [roles] in machineConfig")
 	return -1
 }
+
+// GetInitMachine accepts a client and clusterID and returns the "init node" machine
+// object for rke2/k3s clusters
+func GetInitMachine(client *rancher.Client, clusterID string) (*v1.SteveAPIObject, error) {
+	logrus.Info("Retrieving secret and identifying machine...")
+	secret, err := secrets.ListSecrets(client, local, fleetNamespace, metav1.ListOptions{
+		LabelSelector: initNodeLabelKey + "=" + True,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	initNodeMachineName := secret.Items[0].ObjectMeta.Labels[machineNameSteveLabel]
```

Comment:
can you add a comment that this will always be ordered and [0] is exactly what we're expecting / the order won't change due to the label you're looking at?